tests/test_download.py

```diff
@@ -96,6 +104,87 @@ def test_download_service(
         assert response.status_code == 400, f"{response.data}"
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "userdata,erc20_enterprise",
+    [(False, False), ("valid", False), ("invalid", False), (False, True)],
+)
+def test_download_multiple_orders(
+    client, publisher_wallet, consumer_wallet, web3, userdata, erc20_enterprise
+):
+    nft_factory_address = get_data_nft_factory_address(web3)
+    asset1 = get_registered_asset(publisher_wallet, erc20_enterprise=erc20_enterprise)
+    service1 = get_first_service_by_type(asset1, ServiceType.ACCESS)
+    mint_100_datatokens(
+        web3, service1.datatoken_address, consumer_wallet.address, publisher_wallet
+    )
+    approve_tokens(
+        web3, service1.datatoken_address, nft_factory_address, 1, consumer_wallet
+    )
+    # create an asset with two services
+    asset2 = get_registered_asset(
+        publisher_wallet, erc20_enterprise=erc20_enterprise, no_of_services=2
+    )
+    service2 = get_first_service_by_type(asset2, ServiceType.ACCESS)
```

Comment:
could you please move hte repeated instructions to a helper function in order to improve readability? I'm talking about get_registered_assrt, get_first_service_by_type, minting and approving, which seems to be copied 3 times.

Reply:
done in https://github.com/oceanprotocol/provider/pull/632/commits/f5c7edbc4bf29880742dcf7375b55a58a0de269e